pkg/cloudprovider/providers/kms_service.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudprovider
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/oauth2/google"
+
+	cloudkms "google.golang.org/api/cloudkms/v1"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/kms"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/gce"
+)
+
+type kmsServiceFactory struct {
+	cloudName      string
+	configFilePath string
+}
+
+// NewKMSServiceFactory creates a CloudKMSServiceFactory which can provide various cloud KMS services.
+func NewKMSServiceFactory(name, configFilePath string) kms.CloudKMSServiceFactory {
```

Comment:
We initially had an anonymous function in kube-apiserver/server.go which would get passed to k8s.io/apiserver to inject the cloudprovider dependency. This was suggested by @sttts to bypass the ugliness of that code.